staging/src/k8s.io/apiserver/pkg/server/genericapiserver_graceful_termination_test.go

```diff
@@ -429,58 +464,82 @@ func TestMuxAndDiscoveryComplete(t *testing.T) {
 }
 
 func TestPreShutdownHooks(t *testing.T) {
-	s := newGenericAPIServer(t, true)
-	doer := setupDoer(t, s.SecureServingInfo)
-
-	preShutdownHookErrCh := make(chan error)
-	err := s.AddPreShutdownHook("test-backend", func() error {
-		// this pre-shutdown hook waits for the requests in flight to drain
-		// and then send a series of requests to the apiserver, and
-		// we expect these series of requests to be completed successfully
-		<-s.lifecycleSignals.InFlightRequestsDrained.Signaled()
-
-		// we send 5 requests, once every second
-		var r result
-		client := newClient(true)
-		for i := 0; i < 5; i++ {
-			r = doer.Do(client, func(httptrace.GotConnInfo) {}, fmt.Sprintf("/echo?message=attempt-%d", i), 1*time.Second)
-			if r.err != nil {
-				break
+	tests := []struct {
+		name   string
+		server func() *GenericAPIServer
+	}{
+		{
+			name: "ShutdownSendRetryAfter is disbaled",
+			server: func() *GenericAPIServer {
+				return newGenericAPIServer(t, false)
```

Comment:
this is aligned with the integration framework, `shutdown-send-retry-after` is not enabled. the test in master does not exercise this use case, this PR adds this, and with the chages introduced by https://github.com/kubernetes/kubernetes/pull/110000 it passes.